src/components/AuthForm.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import { Button } from "react-bootstrap";
+import { auth } from "../firebase";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import { useNavigate } from "react-router";
+
+const AuthForm = (props) => {
+  const [emailInput, setEmailInput] = useState("");
+  const [passwordInput, setPasswordInput] = useState("");
+  const [isNewUser, setIsNewUser] = useState(true);
+  const [errorCode, setErrorCode] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const navigate = useNavigate();
+
+  const handleEmailInputChange = (e) => {
+    setEmailInput(e.target.value);
+  };
+
+  const handlePasswordInputChange = (e) => {
+    setPasswordInput(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const closeAuthForm = () => {
+      setEmailInput("");
+      setPasswordInput("");
+      setIsNewUser(true);
+      setErrorCode("");
+      setErrorMessage(""); 
+      navigate("/");
+    };
+
+    const setErrorState = (error) => {
+      setErrorCode(error.code);
+      setErrorMessage(error.message);
+    };
+
+    if (isNewUser) {
+      createUserWithEmailAndPassword(auth, emailInput, passwordInput)
+        .then(closeAuthForm)
+        .catch(setErrorState);
+    } else {
+      signInWithEmailAndPassword(auth, emailInput, passwordInput)
+        .then(closeAuthForm)
+        .catch(setErrorState);
+    }
+  };
+
+  const toggleNewOrReturningAuth = () => {
+    setIsNewUser(false);
```

Comment:
this function doesn't seem to toggle the state, but only set it to false. Maybe rename the function or check the functionality here